File: app-java/app/src/main/java/com/phunware/android/phunwareproducthomework/storage/InMemoryZipCodeStore.java
package com.phunware.android.phunwareproducthomework.storage;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class InMemoryZipCodeStore implements ZipCodeStore {
    private ArrayList<String> zipCodes = new ArrayList<>();

    @Override
    public List<String> getZipCodes() {
        return Collections.unmodifiableList(zipCodes);
    }

    @Override
    public void addZipCode(String zipCode) {
        zipCodes.add(zipCode);
    }
}

File: app-kotlin/app/src/main/java/com/phunware/android/phunwareproducthomework/storage/InMemoryZipCodeStore.kt
package com.phunware.android.phunwareproducthomework.storage

class InMemoryZipCodeStore : ZipCodeStore {
    private var _zipCodes: MutableList<String> = mutableListOf()

    /**
     * Return a mutable list (defensive copy) of our zip codes
     */
    override fun getZipCodes(): List<String> = _zipCodes.toList()

    override fun addZipCode(zipCode: String) {
        _zipCodes.add(zipCode)
    }
}